lib/ts/station/FSM.ex

```diff
@@ -95,16 +99,19 @@ defmodule Station.FSM do
         :invalid ->
           # If invalid query, remove itinerary
           new_station_data = List.delete_at(station_data, 0)
+          Logger.info(fn -> "Current state='query_rcvd'; Next state='ready'" end)
```

Comment:
all the log messages in `query_rcvd` state shall be of debug type. Their message shall be of the type "query <query_id> is invalid".

Similar messages can be constructed for other atoms under this case structure.